src/GuardClauses/GuardClauseExtensions.cs

```diff
@@ -577,13 +660,19 @@ public static TimeSpan NegativeOrZero([JetBrainsNotNull] this IGuardClause guard
         /// <param name="guardClause"></param>
         /// <param name="input"></param>
         /// <param name="parameterName"></param>
+        /// <param name="message"></param>
         /// <returns><paramref name="input" /> if the value is not out of range.</returns>
         /// <exception cref="InvalidEnumArgumentException"></exception>
-        public static T OutOfRange<T>([JetBrainsNotNull] this IGuardClause guardClause, T input, [JetBrainsNotNull] string parameterName) where T : struct, Enum
+        public static T OutOfRange<T>([JetBrainsNotNull] this IGuardClause guardClause, T input, [JetBrainsNotNull] string parameterName, string message = "") where T : struct, Enum
         {
             if (!Enum.IsDefined(typeof(T), input))
             {
-                throw new InvalidEnumArgumentException(parameterName, Convert.ToInt32(input), typeof(T));
+                if (message == "")
+                {
+                    throw new InvalidEnumArgumentException(parameterName, Convert.ToInt32(input), typeof(T));
+                }
+
```

Comment:
consistency of spacing: sometimes with an empty line, sometimes without

I would recommend with an empty line after the closing brace of the if (I assume that is today the default spacing rule)

Reply:
...